File: lab_2/problem_2/Euclidean.c
#include <stdio.h>

int gcd(int a, int b) {
    if (b == 0) {
        return a;
    }
    return gcd(b, a % b);
}   

int main() {
    int a, b;
    printf("Enter first term: ");
    scanf("%d", &a);
    printf("Enter second term: ");
    scanf("%d", &b);
    printf("The GCD is: %d\n", gcd(a, b));
    return 0;
}

File: lab_2/problem_2/Euclidean.py
def gcd(a, b):
    while b != 0:
        a, b = b, a % b
    return a

# Get input from the user
term1 = int(input("Enter the first number: "))
term2 = int(input("Enter the second number: "))

# Calculate the GCD
result = gcd(term1, term2)

# Print the result
print("The Greatest Common Divisor is:", result)